pkgs/development/tools/language-servers/fish-lsp/default.nix

```diff
@@ -0,0 +1,81 @@
+{
+  lib,
+  stdenv,
+  fetchYarnDeps,
+  fixup-yarn-lock,
+  fetchFromGitHub,
+  yarn,
+  fish,
+  nodejs,
+  ...
+}: let
+  rev = "1.0.6";
+
+  pname = "fish-lsp";
+
+  src = fetchFromGitHub {
+    owner = "ndonfris";
+    repo = "fish-lsp";
+    inherit rev;
+    hash = "sha256-o5WomHED/JaamHkEOHdRJHBkfbrfJ4NbvnD/1q1yYSs=";
+  };
+
+  yarnDeps = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-hmaLWO1Sj+2VujrGD2A+COfVE2D+tCnxyojjq1512K4=";
+  };
+in
+  stdenv.mkDerivation
+  {
+    inherit pname;
+    version = "${builtins.substring 0 6 rev}";
+
+    inherit src;
+
+    nativeBuildInputs = [
+      fish
+      yarn
+      nodejs
+      fixup-yarn-lock
+    ];
+
+    buildInputs = [
+      fish
+      nodejs
+    ];
+
+    dontConfigure = true;
+
+    buildPhase = ''
```

Comment:
If you are overriding `configurePhase`, `buildPhase`, `checkPhase`, `installPhase` or any other phase, you should not forget about explicitly running pre and post-phase hooks like [their](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L953) [original](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1008) [definitions](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1037) [do](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1078).

It is generally expected that an appropriate pre-phase hook (e.g. `preBuild`) hook will run at the beginning of a phase (e.g. `buildPhase`) and post-phase hook (e.g. `postBuild`) will run at the end. Hooks are normally ran as a part of a phase so if you override a phase as a whole, you will need to add `runHook hookName` calls (e.g. `runHook preBuild`) manually.

Having phases run pre/post-phase hooks is important because many setup hooks insert their own code into them – omitting a hook might therefore prevent some setup hooks required for proper functionality of a package from running. Additionally, hooks are often inserted by developers into the package expression and by users when overriding a package using `overrideAttrs`. Not running them can thus cause confusion why their code is not executed.

## Examples

### Before

```nix
  installPhase = ''
    your commands
  '';
```

### After

```nix
  installPhase = ''
    runHook preInstall

    your commands

    runHook postInstall
  '';
```

Source: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/missing-phase-hooks.md

Reply:
Oh yes sorry I completely forgot that all official nixpkgs derivation called pre and post callback. 

Plus it's very practical so sorry to have forget `preBuild` / `postBuild` and `preInstall` / `postInstall`.